src/testdir/test_string_interpolation.vim

```diff
@@ -0,0 +1,123 @@
+" Test for string-interpolations $'${x} ${y.z}'
+
+scriptencoding utf-8
+
+func Test_string_interpolation()
+  call s:basic()
+  call s:appendix()
+  call s:illformed()
+endfunc
+
+func s:basic() abort
```

Comment:
It might be better to make this function global. E.g. `Test_string_interpolation_basic`.
If it is global, you can run a specific test. See:
https://github.com/vim/vim/blob/e28cfb28126c07abd9e9ba04e36b3789433201b6/src/testdir/runtest.vim#L5-L7

Reply:
I'll add it :+1: